### ISSUE TITLE
ValidatorTest#testIsValidDirectoryPathGHSL_POC fails on Mac

### ISSUE THREAD

**Issue by stevebosman-oc** [ESAPI/esapi-java-legacy#702]
When I run the project tests on a Mac I get a test failure

Occurs on clean develop branch

To Reproduce
* clone repository on a mac
* run `mvn test`

```
[2022-05-19 11:32:09] [IntrusionDetector] [SECURITY FAILURE Anonymous:@unknown -> /ExampleApplication/IntrusionDetector] Invalid directory, not inside specified parent: context=GHSL-2022-008, input=/var/folders/v1/t3hknvpj3b12cylfqjl167480000gq/T/junit6455395578516278762/esapi-test2, parent=/var/folders/v1/t3hknvpj3b12cylfqjl167480000gq/T/junit6455395578516278762/sibling-of-esapi-test2 
[2022-05-19 11:32:09] [IntrusionDetector] [SECURITY FAILURE Anonymous:@unknown -> /ExampleApplication/IntrusionDetector] Failure to validate directory path: context=GHSL-2022-008, input=/var/folders/v1/t3hknvpj3b12cylfqjl167480000gq/T/junit6455395578516278762/esapi-test2 
[2022-05-19 11:32:09] [IntrusionDetector] [SECURITY FAILURE Anonymous:@unknown -> /ExampleApplication/IntrusionDetector] Invalid directory name does not match the canonical path: context=GHSL-2022-008, input=/var/folders/v1/t3hknvpj3b12cylfqjl167480000gq/T/junit6455395578516278762/sibling-of-esapi-test2/child, canonical=/private/var/folders/v1/t3hknvpj3b12cylfqjl167480000gq/T/junit6455395578516278762/sibling-of-esapi-test2/child 
[2022-05-19 11:32:09] [IntrusionDetector] [SECURITY FAILURE Anonymous:@unknown -> /ExampleApplication/IntrusionDetector] Failure to validate directory path: context=GHSL-2022-008, input=/var/folders/v1/t3hknvpj3b12cylfqjl167480000gq/T/junit6455395578516278762/sibling-of-esapi-test2/child 

java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at org.owasp.esapi.reference.ValidatorTest.testIsValidDirectoryPathGHSL_POC(ValidatorTest.java:417)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
```

**Expected behavior**
All tests pass

**Platform environment:**
 - OS: macOs Monterey
 - Browser N/A
 - Azul Zulu community 1.8

**Additional context**
I think the new assertion at the bottom of the test needs to either be excluded on macOS, or to create a temporary folder in an alternate way on MacOs - since the Javadoc for isValidDirectoryPath states:

> **Note** On platforms that support symlinks, this function will fail canonicalization if directorypath
> is a symlink. For example, on MacOS X, /etc is actually /private/etc. If you mean to use /etc, use its real
> path (/private/etc), not the symlink (/etc).

In this case the temporary folder is like:
    /var/folders/v1/t3hknvpj3b12cylfqjl167480000gq/T/junit6455395578516278762/sibling-of-esapi-test2/child
but the canonical path of this is in private
    /private/var/folders/v1/t3hknvpj3b12cylfqjl167480000gq/T/junit6455395578516278762/sibling-of-esapi-test2/child

**Comment by stevebosman-oc** [comment]
I am happy to look into providing a PR

**Comment by stevebosman-oc** [comment]
I think it may be as simple as changing line 408 to use getCanonicalPath and not getAbsolutePath

**Comment by kwwall** [comment]
Closed per PR #703.
